java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java

```diff
@@ -60,12 +52,17 @@
 import net.osmand.util.MapUtils;
 import rtree.RTree;
 
+import javax.annotation.Nonnull;
+
 public class OsmGpxWriteContext {
+	public static final int POI_SEARCH_POINTS_DISTANCE_M = 5000; // store segments as POI-points every 5 km (POI-search)
+	public static final String OSM_TAG_PREFIX = "osm_";
 	private final static NumberFormat latLonFormat = new DecimalFormat("0.00#####", new DecimalFormatSymbols());
 	public final QueryParams qp;
 	public int tracks = 0;
 	public int segments = 0;
-	long id = -10;
+	private final long baseIdMask = ((1 << 30) - 1); // 30 bits
+	private long baseOsmId = -(new Random(System.currentTimeMillis()).nextLong() & baseIdMask); // 30 bits negative
```

Comment:
Base id is not needed at all, all negative ids are considered custom and never intersect

Reply:
removed